### ISSUE TITLE
[filezilla] Failed to build

### ISSUE THREAD

**Issue by uberhacker** [skycocker/chromebrew#4870]
```
checking for wxWidgets version >= 3.0.4 (--unicode=yes --universal=no)... no
configure: error: 
    The requested wxWidgets build couldn't be found.
    
    The configuration you asked for FileZilla requires a wxWidgets
    build with the following settings:
        --unicode=yes --universal=no
    but such build is not available.

    To see the wxWidgets builds available on this system, please use
    'wx-config --list' command. To use the default build, returned by
    'wx-config --selected-config', use the options with their 'auto'
    default values.

    If you still get this error, then check that 'wx-config' is
    in path, the directory where wxWidgets libraries are installed
    (returned by 'wx-config --libs' command) is in LD_LIBRARY_PATH
    or equivalent variable and wxWidgets version is 3.0.4 or above.
filezilla failed to build: `./configure --prefix=/usr/local --libdir=/usr/local/lib --mandir=/usr/local/share/man --build=armv7l-cros-linux-gnueabihf --host=armv7l-cros-linux-gnueabihf --target=armv7l-cros-linux-gnueabihf --disable-maintainer-mode --with-pugixml=builtin` exited with 1
```

**Comment by satmandu** [comment]
Wxwidgets is installed?

**Comment by uberhacker** [comment]
Yep, I even tried reinstalling from source.

**Comment by vincowl** [comment]
I may have a related problem while trying to install filezilla. The configure step crashes claiming for a missing gtk+-3.0 component that is required by wxwidgets. Wxwidgets, gtk3 and gtk3mm are installed. Problem occurs since the latest version bump.

**Comment by satmandu** [comment]
Does a recompile of gtk3 from source first help?

GCC 10? GCC 10.2.1?

**Comment by satmandu** [comment]
Hmm. Just thinking out loud... 

Maybe we  need a way of embedding the package file used to build a package within the package.

Alongside some amount of other metadata to describe how the package was built.

Maybe it doesn't even need to be in the package file. Maybe it could automatically upload to a web form.

But compiler version, list of installed packages (and those versions), PATH, and environment... Those would all be useful for helping debug these issues when they happen, and they're happening a lot no? 

Build replicability is quite the issue. Probably doesn't help that we're building on top of a moving target...

{c,b}rew is the avowed non-chroot solution, but we are really close to a chroot, no?  If we had a proper chroot we could have a standardized setup (and also use a newer glibc more easily.)

We would just need a way to do a reverse chroot or whatever to get back to the built-in dev system when necessary, right?

(Yes this sounds very similar to what Google has done with their VMs or crosby... Or like what Crouton does... But with more of an emphasis on being closer to bare metal.)

**Comment by uberhacker** [comment]
Not sure I understand what you are suggesting.  I attempted to reinstall gtk3 from source and still no luck.  I also tried to build [wxwidgets from source](#4873).

**Comment by supechicken666** [comment]
> {c,b}rew is the avowed non-chroot solution, but we are really close to a chroot, no? If we had a proper chroot we could have a standardized setup (and also use a newer glibc more easily.)
> 
> We would just need a way to do a reverse chroot or whatever to get back to the built-in dev system when necessary, right?

Do you mean creating minimum system with Chromebrew and something like `busybox` to provide some basic unix commands to run in chroot? If so, this will be more likely to creating a new Linux distribution  :)

**Comment by satmandu** [comment]
@supechicken666 @uberhacker Sorry was taking a family break for the holidays since the partner had time off. :)

And yeah @supechicken666 something minimal like that.

The reason I am suggesting a minimal variant is that our environment is sometimes sharing stuff with the google libraries, and sometimes not... and that's probably a huge source of our issues, right?

But in actuality we're actually using a very very small fraction of the google installed libraries, and that's an annoyance because we're hampered by using an outdated libc (and yes kernel) which are likely causing issues with getting software to work. 

For instance, I suspect my llvm testing is running up against libc version issues. Which might be tied to the older glibc ChromeOS uses?

We may be the only ones trying to get newer software working with these older library versions...

If we did a pivot root to a chroot early on with a newer glibc, and then did all of our testing based upon that, we would be free of being tied to a particular ChromeOS build for most things. This is of course what Crouton does, with the exception of their support not fully integrating wayland, xwayland, or sommelier support.

But imagine also a slimmed down chromebrew tuned just for those things which DO require linking with the ChromeOS libraries, but also tied to them as little as possible to prevent breakage.

You could use a static linked coreutils via musl for instance to limit breakage. There is a strong viable use case for mucking about with ChromeOS to do things like play around with kernel extensions, crosvm, etc where you can't be siloed into a chroot or VM. Having just that would be incredibly useful. Otherwise every package compiled on slightly different hardware referencing slightly differently compiled libraries from google has the possibility of creating replicable package headaches.

Example of working static linked coreutils:

```
require 'package'

class Coreutils_static < Package
  description 'GNU coreutils built statically with musl'
  homepage 'http://www.gnu.org/software/coreutils/coreutils.html'
  version '8.32'
  compatibility 'all'
  source_url 'https://mirrors.sarata.com/gnu/coreutils/coreutils-8.32.tar.xz'
  source_sha256 '4458d8de7849df44ccab15e16b1548b285224dbba5f08fac070c1c0e0bcc4cfa'


  def self.build
  # static build script from https://github.com/luciusmagn/coreutils-static/blob/master/build.sh
system 'cat <<"EOF"> build.sh
#!/bin/bash
#
# build static coreutils because we need exercises in minimalism
# MIT licensed: google it or see robxu9.mit-license.org.
#
# For Linux, also builds musl for truly static linking.

coreutils_version="8.32"
musl_version="1.2.1"

platform=$(uname -s)

if [ -d build ]; then
  echo "= removing previous build directory"
  rm -rf build
fi

#mkdir build # make build directory
#pushd build

# download tarballs
#echo "= downloading coreutils"
#curl -LO http://ftp.gnu.org/gnu/coreutils/coreutils-${coreutils_version}.tar.xz

#echo "= extracting coreutils"
#tar xJf coreutils-${coreutils_version}.tar.xz

if [ "$platform" = "Linux" ]; then
  echo "= downloading musl"
  curl -LO http://www.musl-libc.org/releases/musl-${musl_version}.tar.gz

  echo "= extracting musl"
  tar -xf musl-${musl_version}.tar.gz

  echo "= building musl"
  working_dir=$(pwd)

  install_dir=${working_dir}/musl-install

  pushd musl-${musl_version}
  env CFLAGS="$CFLAGS -Os -ffunction-sections -fdata-sections -fPIC" LDFLAGS=\'-Wl,--gc-sections\' ./configure --prefix=${install_dir}
  make install
  popd # musl-${musl-version}

  echo "= setting CC to musl-gcc"
  export CC=${working_dir}/musl-install/bin/musl-gcc
  export CFLAGS="-static"
else
  echo "= WARNING: your platform does not support static binaries."
  echo "= (This is mainly due to non-static libc availability.)"
fi

echo "= building coreutils"

#pushd coreutils-${coreutils_version}
env FORCE_UNSAFE_CONFIGURE=1 CFLAGS="$CFLAGS -Os -ffunction-sections -fdata-sections -fPIC" LDFLAGS=\'-Wl,--gc-sections\' ./configure
make
#popd # coreutils-${coreutils_version}

popd # build

if [ ! -d releases ]; then
  mkdir releases
fi

echo "= striptease"
strip -s -R .comment -R .gnu.version --strip-unneeded build/coreutils-${coreutils_version}/coreutils
#echo "= compressing"

shopt -s extglob
# upx currently breaks the binaries
#for file in build/coreutils-${coreutils_version}/src/!(*.*)
#do
#	upx --ultra-brute $file
#done
echo "= extracting coreutils binary"
cp src/!(*.*) releases
echo "= done"
EOF'
    system 'chmod +x ./build.sh'
    system './build.sh'
    system 'curl -Lf https://salsa.debian.org/debian/debianutils/-/raw/master/which?inline=false > releases/which && chmod +x releases/which'
    system "sed -i 's#/bin/sh#/usr/local/bin/sh#g' releases/which"

  end

  def self.install
    system "mkdir -p #{CREW_DEST_PREFIX}/bin"
    FileUtils.cd("#{CREW_DEST_PREFIX}/bin") do
      system "echo '#!/bin/bash' > arch"
      system "echo '#{CREW_PREFIX}/bin/uname -m' >> arch"
      system 'chmod +x arch'
    end
    system "cp releases/* #{CREW_DEST_PREFIX}/bin/"
  end
end
```

**Comment by uberhacker** [comment]
Fixed by #5015.
